src/motion_input_shaping/zero_vibration_shaper.py

```diff
@@ -9,57 +9,21 @@
 
 import math
 import numpy as np
+from plant import Plant
 
 
 class ZeroVibrationShaper:
     """A class for implementing zero vibration input shaping theory."""
 
-    def __init__(self, resonant_frequency: float, damping_ratio: float) -> None:
+    def __init__(self, plant: Plant) -> None:
         """
         Initialize the class.
 
-        :param resonant_frequency: The target vibration resonant frequency in Hz.
-        :param damping_ratio: The target vibration damping ratio.
+        :param plant: The Plant instance defining the system that the shaper is targeting.
         """
-        self.resonant_frequency = resonant_frequency
-        self.damping_ratio = damping_ratio
+        self.plant = plant
         self.n = 1  # How many periods to wait before starting deceleration.
```

Comment:
The getter and setter of `self.n` manipulates `self._n`, but all the other functions utilizes `self.n`.  Should all the other functions use `self._n` instead?

Reply:
I have made this change. I also got rid of the property setter since there is no need to modify `n` outside of the class and the error checking is not necessary with how it is computed.